### PR TITLE
[FIX] point_of_sale: customer screen using iot-connected display

### PR DESCRIPTION
**Pull request by caburj**
When selecting the option "IoT-connected display" for customer display, and pos_iot is installed, there is no way from the res.config.settings form to specify the `proxy_ip` field which is used to send the customer display information. Because of this, using IoT-connected display for customer screen doesn't work.

This commit introduces a way to infer the proxy_ip from the specify display device.

Related: https://github.com/odoo/enterprise/pull/64360

### CODE REVIEW

File: addons/point_of_sale/models/pos_config.py [hunk 1]
```diff
@@ -810,13 +810,14 @@ def update_customer_display(self, order, access_token):
 
     def _get_customer_display_data(self):
         self.ensure_one()
+        display_device = self.iot_device_ids.filtered(lambda d: d.type == 'display')
```
Comment (vlst-odoo) [inline addons/point_of_sale/models/pos_config.py:813]: in community this should be just `self.proxy_ip`.
Maybe we can have something like `_get_customer_display_proxy_ip(self)` ( like we have for the frontend )

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/navbar/sale_details_button/sales_detail_report.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
                         <div class="responsive-price">
                             <t t-esc="line['product_name'].substr(0,20)" />
                             <span class="pos-receipt-right-align">
-                                <t t-esc="Math.round(line.qty * Math.pow(10, pos.models['decimal.precision'].find((dp) => dp.name === 'Product Unit of Measure')?.digits)) / Math.pow(10, pos.models['decimal.precision'].find((dp) => dp.name === 'Product Unit of Measure')?.digits)" />
+                                <t t-esc="Math.round(line.quantity * Math.pow(10, pos.models['decimal.precision'].find((dp) => dp.name === 'Product Unit of Measure')?.digits)) / Math.pow(10, pos.models['decimal.precision'].find((dp) => dp.name === 'Product Unit of Measure')?.digits)" />
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/sale_details_button/sales_detail_report.xml:55]: this seems like it should be in a different commit

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/navbar/sale_details_button/sales_detail_report.xml (hunk 2 above)
Comment (caburj) [reply addons/point_of_sale/static/src/app/navbar/sale_details_button/sales_detail_report.xml:55]: yes it is

Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/navbar/sale_details_button/sales_detail_report.xml:55]: :facepalm: my bad, sorry

**Review by vlst-odoo — APPROVED** [review]

File: addons/point_of_sale/models/pos_config.py (hunk 1 above)
Comment (caburj) [reply addons/point_of_sale/models/pos_config.py]: I introduced `_get_display_device_ip` which is overridden in pos_iot.

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Comment by vlst-odoo** [comment]
robodoo rebase-ff
